src/main/java/org/spongepowered/common/mixin/core/authlib/MixinProperty.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.core.authlib;
+
+import com.google.common.base.Objects;
+import com.mojang.authlib.properties.Property;
+import org.spongepowered.api.profile.property.ProfileProperty;
+import org.spongepowered.asm.mixin.Implements;
+import org.spongepowered.asm.mixin.Interface;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+@Mixin(value = Property.class, remap = false)
+@Implements(@Interface(iface = ProfileProperty.class, prefix = "property$"))
+public abstract class MixinProperty {
+
+    @Shadow public abstract String getName();
+    @Shadow public abstract String getValue();
+    @Nullable @Shadow public abstract String getSignature();
+
+    public Optional<String> property$getSignature() {
+        return Optional.ofNullable(this.getSignature());
+    }
+
+    @Override
+    public boolean equals(@Nullable Object other) {
+        if (this == other) {
+            return true;
+        }
+
+        if (other == null || !(other instanceof ProfileProperty)) {
+            return false;
+        }
+
+        ProfileProperty that = (ProfileProperty) other;
+        return Objects.equal(this.getName(), that.getName())
+                && Objects.equal(this.getValue(), that.getValue())
+                && Objects.equal(this.property$getSignature(), that.getSignature());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(this.getName(), this.getValue(), this.property$getSignature());
+    }
+
+    @Override
+    public String toString() {
+        return Objects.toStringHelper(this)
```

Comment:
Unless you want this to print out as `Property`, I'd recommend actually using the API's defined name as `"ProfileProperty"`